src/hooks/useFetchListCallback.ts

```diff
@@ -1,39 +1,32 @@
+import { Web3Provider } from '@ethersproject/providers'
 import { nanoid } from '@reduxjs/toolkit'
 import { TokenList } from '@uniswap/token-lists'
-import useActiveWeb3React from 'hooks/useActiveWeb3React'
+import { networkHooks } from 'connectors'
 import getTokenList from 'lib/hooks/useTokenList/fetchTokenList'
 import resolveENSContentHash from 'lib/utils/resolveENSContentHash'
 import { useCallback } from 'react'
 import { useAppDispatch } from 'state/hooks'
 
-import { getNetworkLibrary } from '../connectors'
 import { fetchTokenList } from '../state/lists/actions'
 
+const ensResolver = async (ensName: string, provider: Web3Provider | undefined, isActive: boolean) => {
+  if (!provider || !isActive) {
+    throw Error('network is inactive')
+  }
+  return resolveENSContentHash(ensName, provider)
+}
+
 export function useFetchListCallback(): (listUrl: string, sendDispatch?: boolean) => Promise<TokenList> {
-  const { chainId, library } = useActiveWeb3React()
   const dispatch = useAppDispatch()
-
-  const ensResolver = useCallback(
-    async (ensName: string) => {
-      if (!library || chainId !== 1) {
-        const networkLibrary = getNetworkLibrary()
-        const network = await networkLibrary.getNetwork()
-        if (networkLibrary && network.chainId === 1) {
-          return resolveENSContentHash(ensName, networkLibrary)
-        }
-        throw new Error('Could not construct mainnet ENS resolver')
-      }
-      return resolveENSContentHash(ensName, library)
-    },
-    [chainId, library]
-  )
+  const provider = networkHooks.useProvider()
```

Comment:
This will not work if the network was switched from MAINNET. There should either be another network provider that _only_ connects to MAINNET (which would allow this to work with no wallet, regardless of the active network), or this should be switched back to MAINNET when a wallet is connected (which would mean this would not work with no wallet connected when not connected to MAINNET).

Reply:
yea you're right, will fix